pkg/file/pipeline/chunkfeeder_writer.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pipeline
+
+import (
+	"encoding/binary"
+)
+
+const span = 8
+
+type chunkFeeder struct {
+	size      int
+	next      ChainWriter
+	buffer    []byte
+	bufferIdx int
+}
+
+func NewChunkFeederWriter(size int, next ChainWriter) Interface {
+	return &chunkFeeder{
+		size:   size,
+		next:   next,
+		buffer: make([]byte, size),
+	}
+}
+
+// Write assumes that the span is prepended to the actual data before the write !
+func (f *chunkFeeder) Write(b []byte) (int, error) {
+	l := len(b) // data length
+	w := 0      // written
+
+	if l+f.bufferIdx < f.size {
+		// write the data into the buffer and return
+		n := copy(f.buffer[f.bufferIdx:], b)
+		f.bufferIdx += n
+		return n, nil
+	}
+
+	// if we are here it means we have to do at least one write
+	d := make([]byte, f.size+span)
```

Comment:
Could it be possible to only use `f.buffer`, instead of creating new array each time ?

Reply:
`f.buffer` stores the actual data. we could theoretically do that, but then it means that `f.buffer` must be of size `f.size+span`, and contains the span at the beginning of the data, which means we should also probably constantly marshal the span into the first eight bytes constantly with the data, even on partial writes. this can save this one allocation and probably can also get rid of `bufferIdx` but I feel it is a bit less clear to the reader. i can try to do this when i finish working on encryption and see how it looks